sdk/core/System.ClientModel/src/Options/ClientLoggingOptions.cs

```diff
@@ -0,0 +1,283 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Internal;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// Exposes client options for logging within a <see cref="ClientPipeline"/>.
+/// </summary>
+public class ClientLoggingOptions
+{
+    private bool _frozen;
+
+    private bool? _enableLogging;
+    private bool? _enableMessageLogging;
+    private bool? _enableMessageContentLogging;
+    private int? _messageContentSizeLimit;
+    private ILoggerFactory? _loggerFactory;
+    private PipelineMessageSanitizer? _sanitizer;
+    private ChangeTrackingStringList? _allowedHeaderNames;
+    private ChangeTrackingStringList? _allowedQueryParameters;
+
+    private static readonly HashSet<string> s_defaultAllowedHeaderNames = ["traceparent",
```

Comment:
should we consider adding a comment to keep these in sync with [Azure.Core](https://github.com/Azure/azure-sdk-for-net/blob/d1ccc6281f15cb94cfab3b899c56ea4d7cd87aeb/sdk/core/Azure.Core/src/DiagnosticsOptions.cs#L44) and vice versa?